### PR TITLE
[PTQ][OV] Fix rank zero output tensor for inplace statistics

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

Nodes with zero output ranks raised runtime error because empty `np.array` has `np.float` type and openvino reduction operations require `np.int64`. Now case `rank==0` works as expected.

### Reason for changes

To fix runtime error during statistic collection for models with nodes with target quantize nodes that have `rank==0` output.

### Related tickets

108784

### Tests

New test in openvino `test_model_transformer.py` that checks corner case with a target point node that has `rank==0` output.

### CODE REVIEW

File: nncf/experimental/openvino_native/graph/node_utils.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def get_reduce_op(node: ov.Node, output_port_id: int) -> ov.Node:
         name_output_port_id = output_port_id
         if reduction_axes_ is None:
             partial_shape = get_partial_shape_safe(node, output_port_id)
-            reduction_axes_ = np.array(range(partial_shape.rank.get_length()))
+            reduction_axes_ = np.arange(partial_shape.rank.get_length()).astype(np.int64)
```
Comment (KodiaqQ) [inline nncf/experimental/openvino_native/graph/node_utils.py:158]: Are you sure that every `op` supports `np.int64` type?

Comment (daniil-lyakhov) [reply nncf/experimental/openvino_native/graph/node_utils.py:158]: Yes, it's covered by tests and described [here](https://docs.openvino.ai/latest/openvino_docs_ops_reduction_ReduceMean_1.html) 
```
T_IND : any supported integer type.
```

**Review by KodiaqQ — APPROVED** [review]
